_base/lang.js

```diff
@@ -307,7 +307,7 @@ define(["./kernel", "../has", "../sniff"], function(dojo, has){
 			// summary:
 			//		Returns true if it is a built-in function or some other kind of
 			//		oddball that *should* report as a function but doesn't
-			return it && !lang.isFunction(it) && /\{\s*\[native code\]\s*\}/.test(String(it)); // Boolean
+			return !!it && lang.isFunction(it) && /\{\s*\[native code\]\s*\}/.test(String(it)); // Boolean
```

Comment:
I think Boolean() is the preferred syntax for TS and ES6, but I agree that it doesn't really matter here, and !! is far more commonly used in Dojo 1.x.

I agree with the other points made. I was trying to figure out if the intent was really to change the second condition to be the opposite of what it is today, or if the real issue is to just make it be guaranteed to be truthy?